OpenRA.Game/GameRules/RulesetWatcher.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Timers;
+
+namespace OpenRA.GameRules
+{
+	public class RulesetWatcher : IDisposable
+	{
+		readonly World world;
+		readonly ModData modData;
+		readonly Dictionary<string, string> watchFiles = new();
+		readonly Timer timer;
+		readonly HashSet<string> fileQueue = new();
```

Comment:
Wondering if the key comparer in your dictionaries/sets should also match the filesystem comparer, as noted in other comments.

Reply:
It does not, this was all done under assumption that casings of filenames just match. But you're right, in order to properly support case-insensitive filesystems (on Windows), comparing paths should use correct comparer/comparison.